ts/src/management/tunnelManagementHttpClient.ts

```diff
@@ -817,8 +817,27 @@ export class TunnelManagementHttpClient implements TunnelManagementClient {
             }
         }
 
+        if (!errorMessage && error.response && error.response.status &&
+            error.response.status >= 400 && error.response.status < 500 &&
+            error.response.headers
+        ) {
+            const headers = error.response.headers;
+            const servedBy = headers['X-Served-By'] || headers['x-served-by'];
+            if (!/tunnel-/.test(servedBy)) {
+                // The response did not include either a ProblemDetails body object or a header
+                // confirming it was served by the tunnel service. This check excludes 5xx status
+                // responses which may include non-firwall network infrastructure issues.
+                const requestDomain = new URL(error.config?.url ??
+                    TunnelServiceProperties.production.serviceUri).host;
+                errorMessage = 'The tunnel request resulted in ' +
+                    `${error.response.status} status, but the request ` +
+                    'did not reach the tunnel service. This may indicate the domain ' +
+                    `'${requestDomain}' is blocked by a firewall.`;
+            }
+        }
+
         if (!errorMessage) {
-            if (error?.response) {
+            if (error.response) {
```

Comment:
Is removing the null ? intentional here? Are we sure in this case that the response is not null?

Reply:
This makes sense, thanks!